valhalla/baldr/graphtile.h

```diff
@@ -45,7 +47,7 @@ class tile_getter_t;
 /**
  * Graph information for a tile within the Tiled Hierarchical Graph.
  */
-class GraphTile {
+class GraphTile : public boost::intrusive_ref_counter<GraphTile, boost::thread_unsafe_counter> {
```

Comment:
I think it might be useful to make the constructors private and use a factory function to construct `boost::intrusive_ptr` objects to find uses of `GraphTile` that sidestep the new "official" way to construct them.

Reply:
Done.